orchagent/lagids.lua

```diff
@@ -37,28 +37,48 @@ if op == "add" then
         end
         -- proposed lag id is different than that in database OR
         -- the portchannel does not exist in the database
-        -- If proposed lagid is available, return the same proposed lag id
-        if redis.call("sismember", "SYSTEM_LAG_ID_SET", tostring(plagid)) == 0 then
-            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(plagid))
-            redis.call("srem", "SYSTEM_LAG_ID_SET", tostring(dblagid))
+        -- If proposed lagid is not available, lpop the first availabe ID
+        local index = redis.call("lpos", "SYSTEM_LAG_IDS_FREE_LIST", tostring(plagid))
+        if index == false then
+            if redis.call("llen", "SYSTEM_LAG_IDS_FREE_LIST") <= 0 then
+                return -1
+            end
+            local lagid = redis.call("lpop", "SYSTEM_LAG_IDS_FREE_LIST")
+            redis.call("hset", "SYSTEM_LAG_ID_TABLE", pcname, lagid)
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", lagid)
+            if dblagid then
```

Comment:
this code under "if dblagid then" is duplicated in both if and else

Reply:
They are the same in the if and else. Not able to simply them as one because of "if" case returns "tonumber(lagid)" while the "else" case returns "plagid"